libraries/ts-command-line/src/providers/CommandLineParameterProvider.ts

```diff
@@ -456,6 +486,72 @@ export abstract class CommandLineParameterProvider {
       throw new Error('Command Line Parser Data was already processed');
     }
 
+    // Search for any ambiguous parameters and throw an error if any are found
+    for (const [parserKey, parameterName] of this._ambiguousParameterNamesByParserKey) {
+      if (data[parserKey]) {
+        // Determine if the ambiguous parameter is a short name or a long name, since the process of finding
+        // the non-ambiguous name is different for each.
+        const duplicateShortNameParameters: CommandLineParameter[] | undefined =
+          this._parametersByShortName.get(parameterName);
+        if (duplicateShortNameParameters) {
+          // We also need to make sure we get the non-ambiguous long name for the parameter, since it is
+          // possible for that the long name is ambiguous as well.
+          const nonAmbiguousLongNames: string[] = [];
+          for (const parameter of duplicateShortNameParameters) {
+            const matchingLongNameParameters: CommandLineParameter[] | undefined =
+              this._parametersByLongName.get(parameter.longName);
+            if (!matchingLongNameParameters?.length) {
+              // This should never happen
+              throw new Error(
+                `Unable to find long name parameters for ambiguous short name parameter "${parameterName}".`
+              );
+            }
+            // If there is more than one matching long name parameter, then we know that we need to use the
+            // scoped long name for the parameter. The scoped long name should always be provided.
+            if (matchingLongNameParameters.length > 1) {
+              if (!parameter.scopedLongName) {
+                // This should never happen
+                throw new Error(
+                  `Unable to find scoped long name for ambiguous short name parameter "${parameterName}".`
+                );
+              }
+              nonAmbiguousLongNames.push(parameter.scopedLongName);
+            } else {
+              nonAmbiguousLongNames.push(parameter.longName);
+            }
+          }
+          throw new Error(
```

Comment:
It's not clear from the code, but I'm guessing this error message is reporting an ambiguous short name,
whereas line 544 is reporting an ambiguous long name.

Could you add code comments to clarify this?  It might be helpful for the code
comment to show the error message with example real parameters instead of variables.